domainbed/lib/misc.py

```diff
@@ -283,6 +289,62 @@ def accuracy(network, loader, weights, device, dataset):
 
     return float(compute_acc), float(compute_f1), float(overlap_class_acc), float(non_overlap_class_acc)
 
+def get_tsne_data(network, loader, device, domain, n=100):
+    df = pd.DataFrame({'latent_vector' : [], 'prediction' : [], 
+                       'class' : [], 'domain' : []})
+
+    zs = []
+    ps = []
+    ys = []
+
+    network.eval()
+    with torch.no_grad():
+        i = 0
+        for x, y in loader:
+            x = x.to(device)
+            y = y.to(device)
+            z = network.featurizer(x) # SAGNet uses .network_f for featurizer
```

Comment:
I'll need to validate that this is the best place to collect features for tsne plot generation

Reply:
```
if not isinstance(network, algorithms.SAGNet):
                z = network.featurizer(x) # SAGNet uses .network_f for featurizer
            else:
                z = network.network_f(x) # SAGNet uses .network_f for featurizer

```
Include this and import
`from domainbed import algorithms`